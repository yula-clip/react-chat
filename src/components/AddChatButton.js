import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import AddIcon from '@material-ui/icons/Add';

const styles = theme => ({
  newChatButton: {
    position: 'absolute',
    left: 'auto',
    right: theme.spacing.unit * 3,
    bottom: theme.spacing.unit * 3 + 48, // + bottom navigation
  },
});

const AddChatButton = ({ classes }) => (
  <Button variant="fab" color="primary" aria-label="Add" className={classes.newChatButton}>
    <AddIcon />
  </Button>
);

export default withStyles(styles)(AddChatButton);
