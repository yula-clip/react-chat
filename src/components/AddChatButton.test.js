/* eslint-env jest */
import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import AddChatButton from "./AddChatButton";

const mockProps = {
  onClick: jest.fn(),
  disabled: false
};

describe("<AddChatButton />", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<AddChatButton {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders button", () => {
    const tree = renderer.create(<AddChatButton {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders disabled button", () => {
    const tree = renderer
      .create(<AddChatButton {...mockProps} disabled />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
