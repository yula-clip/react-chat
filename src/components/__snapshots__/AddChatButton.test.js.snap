// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddChatButton /> renders button 1`] = `
<button
  aria-label="Add"
  className="MuiButtonBase-root-78 MuiButton-root-52 MuiButton-fab-69 MuiButton-contained-63 MuiButton-containedPrimary-64 MuiButton-raised-66 MuiButton-raisedPrimary-67 AddChatButton-newChatButton-50"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-53"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-81"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-92"
  />
</button>
`;

exports[`<AddChatButton /> renders disabled button 1`] = `
<button
  aria-label="Add"
  className="MuiButtonBase-root-78 MuiButtonBase-disabled-79 MuiButton-root-52 MuiButton-fab-69 MuiButton-contained-63 MuiButton-containedPrimary-64 MuiButton-raised-66 MuiButton-raisedPrimary-67 MuiButton-disabled-72 AddChatButton-newChatButton-50"
  disabled={true}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="-1"
  type="button"
>
  <span
    className="MuiButton-label-53"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-81"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </span>
</button>
`;
